Mathlib/Data/Polynomial/Degree/Lemmas.lean

```diff
@@ -421,4 +421,30 @@ theorem leadingCoeff_comp (hq : natDegree q ≠ 0) :
 
 end NoZeroDivisors
 
+section Field
+
+variable {K : Type*} [Field K]
+
+/-! Useful lemmas for the "monicization" of a nonzero polynomial `p`. -/
+@[simp] lemma irreducible_mul_C_leadingCoeff_inv {p : K[X]} (hp0 : p ≠ 0) :
+    Irreducible (p * C (leadingCoeff p)⁻¹) ↔ Irreducible p :=
+  irreducible_mul_isUnit <| isUnit_C.mpr <| IsUnit.mk0 _ <|
+    inv_ne_zero <| leadingCoeff_ne_zero.mpr hp0
```

Comment:
This section is actually duplicate with some results at the start of `FieldDivision`. Let me consolidate them here (higher up the import hierarchy).